libdebug/state/thread_context.py

```diff
@@ -8,115 +8,189 @@
-        """Updates the register values."""
-        if not self._needs_register_poll:
-            self._needs_register_poll = True
-            return
+    @property
+    def pid(self: ThreadContext) -> int:
+        """The process ID of the thread."""
+        return self._internal_debugger.process_id
 
-        liblog.debugger("Polling registers for thread %d", self.thread_id)
+    @property
+    def thread_id(self: ThreadContext) -> int:
+        """The thread ID."""
+        return self._thread_id
 
-        self.registers.poll(self)
-        self._dirty = False
+    @property
+    def tid(self: ThreadContext) -> int:
+        """The thread ID."""
+        return self._thread_id
 
-    def _flush_registers(self: ThreadContext) -> None:
-        """Flushes the register values."""
-        liblog.debugger("Flushing registers for thread %d", self.thread_id)
+    @property
+    def running(self: ThreadContext) -> bool:
+        """Whether the process is running."""
+        return self._internal_debugger.running
 
-        if self._dirty:
-            self.registers.flush(self)
+    @property
+    def exit_code(self: ThreadContext) -> int | None:
+        """The thread's exit code."""
+        self._internal_debugger._ensure_process_stopped()
+        if not self.thread_dead:
+            liblog.warning("Thread is not dead. No exit code available.")
+        elif self._exit_code is None and self._exit_signal is not None:
+            liblog.warning(
+                "Thread exited with signal %s. No exit code available.",
+                resolve_signal_name(self._exit_signal),
+            )
+        return self._exit_code
+
+    @property
+    def exit_signal(self: ThreadContext) -> int | None:
+        """The thread's exit signal."""
+        self._internal_debugger._ensure_process_stopped()
+        if not self.thread_dead:
+            liblog.warning("Thread is not dead. No exit signal available.")
+            return None
+        elif self._exit_signal is None and self._exit_code is not None:
+            liblog.warning("Thread exited with code %d. No exit signal available.", self._exit_code)
+            return None
+        return resolve_signal_name(self._exit_signal)
+
+    @property
+    def signal(self: ThreadContext) -> str | None:
+        """The signal will be forwarded to the thread."""
+        self._internal_debugger._ensure_process_stopped()
+        return None if self._signal_number == 0 else resolve_signal_name(self._signal_number)
+
+    @signal.setter
+    def signal(self: ThreadContext, signal: str | int) -> None:
+        """Set the signal to forward to the thread."""
+        self._internal_debugger._ensure_process_stopped()
+        if self._signal_number != 0:
+            liblog.debugger(
+                f"Overwriting signal {resolve_signal_name(self._signal_number)} with {resolve_signal_name(signal) if isinstance(signal, int) else signal}."
+            )
+        if isinstance(signal, str):
```

Comment:
Nope, they are doing the opposite. At line 148, I am converting the signal only if it is an int (I want the mnemonic string). In the if branch, I am converting the str into an int. If I swap the position as suggested, it would need to ALWAYS convert the signal for the logging at line 148.